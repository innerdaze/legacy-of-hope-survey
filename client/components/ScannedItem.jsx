import React, { Component } from 'react'
import PropTypes from 'prop-types'
import ListItem from 'grommet/components/ListItem'
import Box from 'grommet/components/Box'
import Menu from 'grommet/components/Menu'
import Anchor from 'grommet/components/Anchor'
import ActionsIcon from 'grommet/components/icons/base/Actions'
import EditIcon from 'grommet/components/icons/base/Edit'
import TrashIcon from 'grommet/components/icons/base/Trash'

class ScannedItem extends Component {

  render() {
    return (
      <ListItem
        justify='between'
        responsive={false}
        >
        <Box direction='row' basis='1/2' justify='between'>
          <Box>{this.props.productID}</Box>
          <Box>{this.props.title}</Box>
        </Box>
        <Box direction='row' basis='1/2' justify='end' textAlign='right'>
          <Box className='secondary'>{this.props.quantity} x Units</Box>
          <Menu
            icon={<ActionsIcon/>}
            >
            <Anchor
              icon={<EditIcon/>}
              label='Change Quantity'
              onClick={this.props.onChangeQuantityClick}
              />
            <Anchor
              icon={<TrashIcon/>}
              label='Delete Item'
              onClick={this.props.onDeleteClick}
              />
          </Menu>
        </Box>
      </ListItem>
    )
  }
}

ScannedItem.propTypes = {
  title: PropTypes.string.isRequired,
  onChangeQuantityClick: PropTypes.func.isRequired,
  onDeleteClick: PropTypes.func.isRequired,
  productID: PropTypes.string,
  quantity: PropTypes.number.isRequired
}

ScannedItem.defaultProps = {
  productID: null
}

export default ScannedItem
