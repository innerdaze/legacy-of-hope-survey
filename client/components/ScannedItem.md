ScannedItem example:

  <ScannedItem
    productID='0123456789'
    title='Product A'
    quantity={5},
    />
