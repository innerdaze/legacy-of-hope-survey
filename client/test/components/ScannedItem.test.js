import React from 'react'
import ScannedItem from '../../components/ScannedItem.jsx'

test('render a ScannedItem', () => {
  const wrapper = shallow(
    <ScannedItem
      data={{}}
      onChangeQuantityClick={Function.prototype}
      onDeleteClick={Function.prototype}
      render={Function.prototype}
    />
  )
  expect(wrapper).toMatchSnapshot()
})

test('render a ScannedItem with data and a render prop', () => {
  const wrapper = shallow(
    <ScannedItem
      data={{ title: 'TEST' }}
      onChangeQuantityClick={Function.prototype}
      onDeleteClick={Function.prototype}
      render={({ title }) => <span>{title}</span>}
    />
  )
  expect(wrapper).toMatchSnapshot()
  expect(wrapper.find('span').text()).toEqual('TEST')
})
