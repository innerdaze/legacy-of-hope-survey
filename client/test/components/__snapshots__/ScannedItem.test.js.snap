// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a ScannedItem 1`] = `
<ListItem
  align="start"
  direction="row"
  justify="between"
  pad={
    Object {
      "horizontal": "medium",
      "vertical": "small",
    }
  }
  responsive={false}
  role="listitem"
  separator="bottom"
>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    justify="end"
    pad="none"
    responsive={true}
    tag="div"
    textAlign="right"
  >
    <Box
      announce={false}
      className="secondary"
      direction="column"
      focusable={true}
      pad="none"
      responsive={true}
      tag="div"
    >
      1
       x Units
    </Box>
    <Menu
      closeOnClick={true}
      direction="column"
      dropAlign={
        Object {
          "left": "left",
          "top": "top",
        }
      }
      icon={
        <Actions
          responsive={true}
        />
      }
      pad="none"
    >
      <Anchor
        animateIcon={true}
        icon={
          <Edit
            responsive={true}
          />
        }
        label="Change Quantity"
        method="push"
        onClick={[Function]}
        tag="a"
      />
      <Anchor
        animateIcon={true}
        icon={
          <Trash
            responsive={true}
          />
        }
        label="Delete Item"
        method="push"
        onClick={[Function]}
        tag="a"
      />
    </Menu>
  </Box>
</ListItem>
`;

exports[`render a ScannedItem with data and a render prop 1`] = `
<ListItem
  align="start"
  direction="row"
  justify="between"
  pad={
    Object {
      "horizontal": "medium",
      "vertical": "small",
    }
  }
  responsive={false}
  role="listitem"
  separator="bottom"
>
  <span>
    TEST
  </span>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    justify="end"
    pad="none"
    responsive={true}
    tag="div"
    textAlign="right"
  >
    <Box
      announce={false}
      className="secondary"
      direction="column"
      focusable={true}
      pad="none"
      responsive={true}
      tag="div"
    >
      1
       x Units
    </Box>
    <Menu
      closeOnClick={true}
      direction="column"
      dropAlign={
        Object {
          "left": "left",
          "top": "top",
        }
      }
      icon={
        <Actions
          responsive={true}
        />
      }
      pad="none"
    >
      <Anchor
        animateIcon={true}
        icon={
          <Edit
            responsive={true}
          />
        }
        label="Change Quantity"
        method="push"
        onClick={[Function]}
        tag="a"
      />
      <Anchor
        animateIcon={true}
        icon={
          <Trash
            responsive={true}
          />
        }
        label="Delete Item"
        method="push"
        onClick={[Function]}
        tag="a"
      />
    </Menu>
  </Box>
</ListItem>
`;
